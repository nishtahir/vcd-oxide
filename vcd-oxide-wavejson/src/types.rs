use serde::{Serialize, Deserialize};

#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WaveJson {
    pub signal: Vec<Signal>,
    #[serde(skip)]
    pub head: Head,
    #[serde(skip)]
    pub foot: Foot,
}

#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Signal {
    pub name: Option<String>,
    pub wave: Option<String>,
    pub data: Option<String>,
}

#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Head {
}

#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Foot {
}
