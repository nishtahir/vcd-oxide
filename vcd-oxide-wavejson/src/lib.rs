mod types;

use crate::types::WaveJson;

#[cfg(test)]
mod test {

    use insta::assert_debug_snapshot;

    use super::*;

    #[test]
    fn test_deserialize_sample_wavejson() {
        let sample = include_str!("../test/res/simple.json");
        let wave: WaveJson = serde_json::from_str(&sample).unwrap();
        assert_debug_snapshot!(wave);
    }
}
