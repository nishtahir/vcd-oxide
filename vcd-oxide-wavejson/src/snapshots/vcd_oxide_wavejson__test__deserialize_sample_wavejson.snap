---
source: vcd-oxide-wavejson/src/lib.rs
assertion_line: 16
expression: wave
---
WaveJson {
    signal: [
        Signal {
            name: Some(
                "pclk",
            ),
            wave: Some(
                "p.......",
            ),
            data: None,
        },
        Signal {
            name: Some(
                "Pclk",
            ),
            wave: Some(
                "P.......",
            ),
            data: None,
        },
        Signal {
            name: Some(
                "nclk",
            ),
            wave: Some(
                "n.......",
            ),
            data: None,
        },
        Signal {
            name: Some(
                "Nclk",
            ),
            wave: Some(
                "N.......",
            ),
            data: None,
        },
        Signal {
            name: None,
            wave: None,
            data: None,
        },
        Signal {
            name: Some(
                "clk0",
            ),
            wave: Some(
                "phnlPHNL",
            ),
            data: None,
        },
        Signal {
            name: Some(
                "clk1",
            ),
            wave: Some(
                "xhlhLHl.",
            ),
            data: None,
        },
        Signal {
            name: Some(
                "clk2",
            ),
            wave: Some(
                "hpHplnLn",
            ),
            data: None,
        },
        Signal {
            name: Some(
                "clk3",
            ),
            wave: Some(
                "nhNhplPl",
            ),
            data: None,
        },
        Signal {
            name: Some(
                "clk4",
            ),
            wave: Some(
                "xlh.L.Hx",
            ),
            data: None,
        },
    ],
    head: Head,
    foot: Foot,
}
